elife/postgresql-10.sls

```diff
@@ -0,0 +1,154 @@
+# postgresql-10.sls is intended to be a drop-in replacement for postgresql.sls (psql 9.4)
+# the two are mutually exclusive and share many of the same state names
+
+# copied from postgresql-client.sls
+
+{% set oscodename = salt['grains.get']('oscodename') %}
+
+# http://www.postgresql.org/download/linux/ubuntu/
+postgresql-deb:
+    pkgrepo.managed:
+        # http://www.postgresql.org/download/linux/ubuntu/
+        - humanname: Official Postgresql Ubuntu LTS
+        - name: deb http://apt.postgresql.org/pub/repos/apt/ {{ oscodename }}-pgdg main
+        - key_url: https://www.postgresql.org/media/keys/ACCC4CF8.asc
+
+# copied from 
+
+pgpass-file:
+    file.managed:
+        - name: /root/.pgpass
+        - source: salt://elife/config/root-pgpass
+        - template: jinja
+        - mode: 0600
+        - defaults:
+            user: {{ pillar.elife.db_root.username }}
+            pass: {{ pillar.elife.db_root.password }}
+            host: localhost
+            port: 5432
+
+{% if salt['elife.cfg']('cfn.outputs.RDSHost') %}
+pgpass-rds-entry:
+    file.append:
+        - name: /root/.pgpass
+        - source: salt://elife/config/root-pgpass
+        - template: jinja
+        - defaults:
+            user: {{ pillar.elife.db_root.username }}
```

Comment:
shouldn't this be `salt['elife.cfg']('project.rds_username')`? RDS provisioning by builder surely can't know the value of a pillar, even if they will probably coincide (`root` being the value?)

Reply:
I'll take a closer look

Reply:
we seem to have hardcoded the rds username into the CloudFormation template: https://github.com/elifesciences/builder/blob/master/src/buildercore/cfngen.py#L312

and `db_root` is also using the same value: 
https://github.com/elifesciences/builder-private/blob/master/pillar/elife.sls#L191-L198

while we can rely on this value being constant, its the rds password that needs to be pulled from the buildvars

Reply:
also, it's not something I want to be changing in this PR. `postgresql-10.sls` is pretty much a copy+paste of `postgresql.sls` with a few things tweaked.

Reply:
marking this as resolved. Let me know if you want me to address it further

Reply:
Agree this all works in this porting. Would address separately as the pillar can't really be changed at the moment without this breaking, I guess.